pkg/ranchhand/processhosts.go

```diff
@@ -263,3 +246,75 @@ func constrainDockerVersion(client *ssh.Client) error {
 
 	return nil
 }
+
+// install docker onto a new system and mark the operation as complete thereafter
+func installDocker(client *ssh.Client, osInfo *osi.Info) error {
+	if err := createRemoteDir(client, remoteStateDir); err != nil {
+		return errors.Wrap(err, "cannot create remote state dir")
+	}
+
+	indicator := filepath.Join(remoteStateDir, "docker-installed")
+	if _, err := client.ExecuteCmd(fmt.Sprintf("test -f %s", indicator)); err == nil {
+		return nil
+	}
+
+	cmds := append(dockerInstallCmds[osInfo.ID], "sudo usermod -aG docker $USER")
+	log.Infof("installing docker on host [%s]", client.RemoteAddr())
+	if _, err := client.ExecuteCmd(strings.Join(cmds, " && ")); err != nil {
+		return errors.Wrap(err, "docker install failed")
+	}
+
+	err := retry.Retry(
+		func(attempt uint) (err error) {
+			if _, err = client.ExecuteCmd("sudo docker version"); err != nil {
+				log.Warnf("attempt [%d] to verify docker is running failed on host [%s]", attempt, client.RemoteAddr())
+			}
+			return err
+		},
+		strategy.Wait(10*time.Second),
+		strategy.Limit(12),
+	)
+	if err != nil {
+		return errors.Wrap(err, "unable to verify docker install")
+	}
+
+	_, err = client.ExecuteCmd(fmt.Sprintf("sudo touch %s", indicator))
+	return errors.Wrap(err, "cannot mark docker install complete")
+}
+
+// install static k8s config files
+func installK8sConfigs(client *ssh.Client) error {
+	if err := createRemoteDir(client, "/etc/kubernetes"); err != nil {
+		return errors.Wrap(err, "cannot create k8s cfg dir")
+	}
+	if err := createStaticCfg(client, k8sAuditCfg, "/etc/kubernetes/audit.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s audit logging cfg")
+	}
+	if err := createStaticCfg(client, k8sAdmissionCfg, "/etc/kubernetes/admission.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s admission policy cfg")
+	}
+	if err := createStaticCfg(client, k8sEventCfg, "/etc/kubernetes/event.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s event limiting cfg")
+	}
+
+	return nil
+}
+
+func createStaticCfg(client *ssh.Client, contents, filename string) error {
+	if _, cerr := client.ExecuteCmd(fmt.Sprintf("test -f %s", filename)); cerr != nil {
```

Comment:
Is there a case where we'd want these files to be updated too?

Reply:
It seems highly unlikely for these files expect in the case of `/etc/kubernetes/event.yaml`, where we'd want to change the _queries per second_ or _burst_ limits. Regardless, let me look at what it would take to add the _update if different_ bits.